sqlflow/client.py

```diff
@@ -0,0 +1,68 @@
+"""Client for interacting with the SQLFlow Server API."""
+
+import os
+import logging
+import grpc
+import google.protobuf.wrappers_pb2 as wrapper
+
+import sqlflow.proto.sqlflow_pb2 as pb
+import sqlflow.proto.sqlflow_pb2_grpc as pb_grpc
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class Client:
+    def __init__(self, server_url=None):
+        """A minimum client that issues queries to and fetch results/logs from sqlflowserver.
+
+        Args:
+            server_url(str):    sqlflowserver url. If None, read value from environment
+                                variable SQLFLOW_SERVER
+
+        Raises:
+            KeyError:
+                Raised if SQLFLOW_SERVER is not specified as environment variable
+        """
+        if server_url is None:
+            try:
+                server_url = os.environ["SQLFLOW_SERVER"]
+            except KeyError:
+                raise ValueError("Can't find environment variable SQLFLOW_SERVER")
+
+        channel = grpc.insecure_channel(server_url)
+        self._stub = pb_grpc.SQLFlowStub(channel)
+
+    def execute(self, operation):
+        """Run a SQLFlow operation
+
+        Argument:
+            operation(str): SQL query to be executed.
+
+        Returns:
+            Generator: generates the response of the server
+        """
+        for res in self._stub.Run(pb.RunRequest(sql=operation)):
+            if res.WhichOneof('response') == 'messages':
+                for message in res.messages.messages:
+                    yield message
```

Comment:
try code below:
  if res.WhichOneof('response') == 'messages':
    for message in res.messages.messages:
      _LOGGER.info(message)
  else: 
    yield from self._decode_protobuf(res)

  def _decode_protobuf(res):
    // convert columns to rows, and yield each row.
    // We may need change proto to return rows instead of columns.

Reply:
> // convert columns to rows, and yield each row.

In my mind, a client is responsible for hiding grpc connection and protobuf structures. So if it receives a map in protobuf, it returns a dict in python.

> // We may need to change proto to return rows instead of columns.

I am not sure about changing the proto at this point. I think @wangkuiyi and @ktong were strongly advocating column oriented encoding in our early discussion.

Reply:
The data returns from server is column based. However, when shows data to user, It's better to be row based as most user think data is row based